gfx_window.h

```diff
@@ -268,6 +285,26 @@ class GfxWindowInternal
                     gfx_window->resetKeyBindings();
                 }
                 break;
+            case WM_DROPFILES:
+                {
+                    HDROP hdrop = (HDROP)w_param;
+                    if(gfx_window->drop_callback != nullptr)
+                    {
+                        char file_name[MAX_PATH];
+                        // Get the number of files dropped onto window
+                        uint32_t fileCount = DragQueryFileA(hdrop, 0xFFFFFFFF, file_name, MAX_PATH);
+                        for(uint32_t i = 0; i < fileCount; i++)
+                        {
+                            // Get i'th filename
+                            DragQueryFileA(hdrop, i, file_name, MAX_PATH);
+
+                            // Pass to callback function
+                            (*gfx_window->drop_callback)(file_name, i, gfx_window->callback_data);
+                        }
+                    }
+                    DragFinish(hdrop);
+                }
+                break;
```

Comment:
If a window is created using `kGfxCreateWindowFlag_AcceptDrop` then it will allow drag and drop events onto the window. This will trigger a `WM_DROPFILES` event where the wParam contains the files(s) dropped onto the window. As its possible for multiple files to be dropped we must loop through all of them and retrieve the dropped file path using `DragQueryFileA`. The file path is sent through the callback along with the index into the file list and the user specified callback data (which allows passing the this pointer of the user)